libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.WebApi/HttpConfigurationExtensions.cs

```diff
@@ -29,73 +29,63 @@ public static HttpConfiguration MapBotFramework(this HttpConfiguration httpConfi
-            ConfigureBotRoutes(botFrameworkAdapter);
+                // TODO: fix up constructor to take options
+                botFrameworkAdapter = new BotFrameworkAdapter(credentialProvider, options.ConnectorClientRetryPolicy, options.HttpClient);
+            }
 
-            return httpConfiguration;
+            // error handler
+            botFrameworkAdapter.ErrorHandler = options.ErrorHandler;
 
-            BotFrameworkAdapter GetOrCreateBotFrameworkAdapter()
+            // add middleware
+            foreach (var middleware in options.Middleware)
             {
-                if (!(httpConfiguration.DependencyResolver.GetService(typeof(BotFrameworkAdapter)) is BotFrameworkAdapter adapter))
-                {
-                    var credentialProvider = ResolveCredentialProvider();
+                botFrameworkAdapter.Use(middleware);
+            }
 
-                    adapter = new BotFrameworkAdapter(credentialProvider, options.ConnectorClientRetryPolicy, options.HttpClient);
-                }
+            ConfigureBotRoutes(httpConfiguration, options, botFrameworkAdapter);
 
-                return adapter;
-            }
+            return httpConfiguration;
+        }
 
-            void ConfigureMiddleware(BotFrameworkAdapter adapter)
-            {
-                foreach (var middleware in options.Middleware)
-                {
-                    adapter.Use(middleware);
-                }
-            }
+        private static void ConfigureBotRoutes(HttpConfiguration httpConfiguration, BotFrameworkOptions options, BotFrameworkAdapter adapter)
+        {
+            var routes = httpConfiguration.Routes;
+            var baseUrl = options.Paths.BasePath;
 
-            void ConfigureBotRoutes(BotFrameworkAdapter adapter)
+            if (options.EnableProactiveMessages)
             {
-                var routes = httpConfiguration.Routes;
-                var baseUrl = options.Paths.BasePath;
-
-                if (!baseUrl.EndsWith("/"))
-                {
-                    baseUrl += "/";
-                }
-
-                if (options.EnableProactiveMessages)
-                {
-                    routes.MapHttpRoute(
-                        BotProactiveMessageHandler.RouteName,
-                        baseUrl + options.Paths.ProactiveMessagesPath,
-                        defaults: null,
-                        constraints: null,
-                        handler: new BotProactiveMessageHandler(adapter));
-                }
-
                 routes.MapHttpRoute(
-                        BotMessageHandler.RouteName,
-                        baseUrl + options.Paths.MessagesPath,
-                        defaults: null,
-                        constraints: null,
-                        handler: new BotMessageHandler(adapter));
+                    BotProactiveMessageHandler.RouteName,
+                    baseUrl.Trim('/') + "/" + options.Paths.ProactiveMessagesPath.Trim('/'),
```

Comment:
Not sure why we would want to trim the trailing / off a path if that's what they wanted it to be.